File: c++/Lab5(31)/Lab5(31)/TBall.cpp
#include <iostream>
#define _USE_MATH_DEFINES 
#include <math.h>
#include "TBall.h";
using namespace std;

TBall::TBall() {
	this->r = 1 + (double)rand() / RAND_MAX * 50;
}
double TBall::getRadius() {
	return this->r;
}
void TBall::SetRadius() {
	this->r = r;
}
double TBall::square_of_surface() {
	double s = 4 * M_PI * this->r * this->r;
	return s;
}
double TBall::volume() {
	double v = this->r * this->r * this->r;
	v = v * M_PI * 0.75;
	return v;
}

File: python/Lab_5/Lab_5/TBall.py
import Tbody as f
import random as r
import math
class TBall(f.TBody):
    def __init__(self):
        self.r=1 + (r.random() * (50 - 1))
    def getRadius(self):
        return self.r
    def SerRadius(self,r):
        self.r=r
    def square_of_surface(self):
        s=4*math.pi*self.r*self.r
        return s
    def volume(self):
        v=self.r*self.r*self.r*math.pi*3/4
        return v

